### DISCUSSION TITLE
boxShadow assignment via c#

### DISCUSSION THREAD

**Discussion by silent21century** [AvaloniaUI/Avalonia#15698] [Q&A] [answered]
i have a case where i need to set the shadow of a border via c#, i looked around on how to do so but i can't seem to get any example i have tried the following which is correct syntax in my editor but the app crashes everytime, 

```cs
// i tried this first 
pocketBorder.BoxShadow = new BoxShadows(
   new BoxShadow{
      Color=Colors.Black, 
      Blur=2, 
      OffsetX=0, 
      OffsetY=0, 
      Spread=1
      }
);
// this also didn't work
pocketBorder.BoxShadow = BoxShadows.Parse("0 0 2 1 Black");

```
the exception returns this message " Object reference not set to an instance of an object. "

**Comment by maxkatz6** [comment]
Can you add stack trace of the exception?

**Reply by silent21century** [reply]
i'm actually using avalonia in vscode which lacks stack trace when crashing, apart from that my issue above seemed to be caused by 'Checked' attribute of ToggleSwitch, it was running the event before the app could initialized the control 'pocketBorder', thanks for tuning in

**Reply by timunie** [reply, marked as answer]
it should show a stack trace if you do some logging. That isn't limited to the IDE. 

In any case: 
- Make sure control is yet loaded before accessing it
- Make sure you are on the UIThread and not in an async task.

**Reply by maxkatz6** [reply]
Last time I checked, VSCode had stacktraces on debugger tab.

**Reply by silent21century** [reply]
oh yeah my bad, i played wrong with my settings before but now its back to normal thanks for mentioning that i thought vscode was limited
